File: backend/src/InsuraTech.API/Controllers/TravelPlansController.cs
using InsuraTech.Application.TravelPlans.DTOs;
using InsuraTech.Application.TravelPlans.Queries.CalculateTravelPlan;
using InsuraTech.Domain.Policies.TravelPlan;
using MediatR;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;

namespace InsuraTech.API.Controllers;

[ApiController]
[Authorize]
[Route("api/v1/travel-plans")]
[Produces("application/json")]
public sealed class TravelPlansController : ControllerBase
{
    private readonly IMediator _mediator;

    public TravelPlansController(IMediator mediator) => _mediator = mediator;

    /// <summary>Calcula el valor de una póliza de viaje dado el tipo, continente y duración.</summary>
    [HttpGet("calculate")]
    [ProducesResponseType(typeof(TravelPlanCalculationDto), StatusCodes.Status200OK)]
    [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status422UnprocessableEntity)]
    [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status503ServiceUnavailable)]
    public async Task<IActionResult> Calculate(
        [FromQuery] TripType tripType,
        [FromQuery] Continent? continent,
        [FromQuery] int durationDays,
        CancellationToken cancellationToken)
    {
        var result = await _mediator.Send(
            new CalculateTravelPlanQuery
            {
                TripType    = tripType,
                Continent   = continent,
                DurationDays = durationDays
            },
            cancellationToken);

        return Ok(result);
    }
}

File: backend_v2/src/main/java/com/insuratech/api/controllers/TravelPlansController.java
// Spec: SPEC-015 — planes de viaje
package com.insuratech.api.controllers;

import com.insuratech.application.travelplans.TravelPlanService;
import com.insuratech.application.travelplans.dto.TravelPlanCalculationDto;
import com.insuratech.domain.policies.travel.Continent;
import com.insuratech.domain.policies.travel.TripType;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/v1/travel-plans")
@RequiredArgsConstructor
public class TravelPlansController {

    private final TravelPlanService travelPlanService;

    @GetMapping("/calculate")
    public ResponseEntity<TravelPlanCalculationDto> calculate(
            @RequestParam TripType tripType,
            @RequestParam(required = false) Continent continent,
            @RequestParam int durationDays) {
        return ResponseEntity.ok(travelPlanService.calculate(tripType, continent, durationDays));
    }
}
